### PR TITLE
[W4.3d][F10-A3]Tan Zhen Yong

### PR DESCRIPTION
**Pull request by Xenonym**

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -26,7 +31,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+
+        String[] splitAddress = trimmedAddress.split(", ");
+        block = new Block(splitAddress[0]);
+        street = new Street(splitAddress[1]);
+        unit = new Unit(splitAddress[2]);
+        postalCode = new PostalCode(splitAddress[3]);
```
Comment (whipermr5) [inline src/seedu/addressbook/data/person/Address.java]: The indices are magic numbers; perhaps they can be extracted out.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -38,22 +48,62 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return String.format("%s, %s, %s, %s", block.get(), street.get(), unit.get(), postalCode.get());
```
Comment (whipermr5) [inline src/seedu/addressbook/data/person/Address.java]: Consider overriding `toString` in each of the inner classes - this way, `.get()` can be omitted here.

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -8,11 +8,16 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should be in the following format: " +
+            "BLOCK, STREET, UNIT, POSTAL_CODE";
+    public static final String ADDRESS_VALIDATION_REGEX = "\\d+, [\\w\\d\\s]+, #\\d+-\\d+, \\d{6}$";
```
Comment (whipermr5) [inline src/seedu/addressbook/data/person/Address.java]: Consider adding a comment to explain the regex. Alternatively, break the regex up into a few parts, extract the parts as constants and give them explanatory names.

**Comment by whipermr5** [comment]
@Xenonym Good job! Some comments added. Please close the PR after reading comments.

**Comment by Xenonym** [comment]
Thanks for the comments! Will be closing this PR.

**Pull request closed without merging**